proxy_ring_buffer.go

```diff
@@ -0,0 +1,278 @@
+		return ReadResultSetSequenceUnavailable, err
+	}
+	return codec.DecodeRingbufferRemainingCapacityResponse(response), nil
+}
+
+// ReadMany
+// Reads a batch of items from the Ringbuffer. If the number of available items after the first read item is smaller
+// than the maxCount, these items are returned. So it could be the number of items read is smaller than the maxCount.
+// If there are fewer items available than minCount, then this call blacks. Reading a batch of items is likely to
+// perform better because less overhead is involved. A filter can be provided to only select items that need to be read.
+// If the filter is null, all items are read. If the filter is not null, only items where the filter function returns
+// true are returned. Using filters is a good way to prevent getting items that are of no value to the receiver.
+// This reduces the amount of IO and the number of operations being executed, and can result in a significant performance improvement.
+func (rb *Ringbuffer) ReadMany(ctx context.Context, startSequence int64, minCount int32, maxCount int32, filter interface{}) (ReadResultSet, error) {
+	if filter != nil {
+		return ReadResultSet{}, errors.New("filter functions are not yet supported in th Go client, please use 'nil' for now")
+	}
+	request := codec.EncodeRingbufferReadManyRequest(rb.name, startSequence, minCount, maxCount, nil)
+	response, err := rb.invokeOnPartition(ctx, request, rb.partitionID)
+	if err != nil {
+		return ReadResultSet{}, err
+	}
+	readCount, items, itemSeqs, nextSeq := codec.DecodeRingbufferReadManyResponse(response)
+	return ReadResultSet{
+		rb:        rb,
+		readCount: readCount,
+		items:     items,
+		itemSeqs:  itemSeqs,
+		nextSeq:   nextSeq,
+	}, nil
+}
+
+// ReadCount Number of items that have been read before filtering.
+func (rrs *ReadResultSet) ReadCount() int32 {
+	return rrs.readCount
+}
+
+// Get one item from List of items that have been read.
+func (rrs *ReadResultSet) Get(index int) (interface{}, error) {
+	if err := rrs.validateIndexInRange(index); err != nil {
+		return nil, err
+	}
+	return rrs.rb.convertToObject(rrs.items[index])
+}
+
+// GetSequence one sequence number from List of sequence numbers for the items that have been read.
+func (rrs *ReadResultSet) GetSequence(index int) (int64, error) {
+	if err := rrs.validateIndexInRange(index); err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	return rrs.itemSeqs[index], nil
+}
+
+// Size the total size of
+func (rrs *ReadResultSet) Size() int {
+	return len(rrs.items)
+}
+
+// GetNextSequenceToReadFrom sequence number of the item following the last read item.
+func (rrs *ReadResultSet) GetNextSequenceToReadFrom() int64 {
+	return rrs.nextSeq
+}
+
+func (rrs *ReadResultSet) validateIndexInRange(index int) error {
+	if index < 0 || int32(index) >= rrs.readCount {
+		return hzcerrors.NewIllegalArgumentError(fmt.Sprintf("index out of range [%d] with length %d", index, rrs.readCount), nil)
+	}
+	return nil
+}
```

Comment:
You can use `check.WithinRangeInt32(index, 0, rrs.readCount - 1)` instead.

Reply:
Done.
I did simplified further and inlined the single line check, as it did add no further value IMHO
fe1b3c0d4b5d741111ae54a31d568fdbb7f8a506